### PR TITLE
Moved "MUST NOT" from app-limited section to main cubic AI section

### PR DESCRIPTION
**Pull request by bbriscoe**
Intended to resolve NTAP/rfc8312bis/#97

### CODE REVIEW

**Review by larseggert — APPROVED** [review]

File: draft-ietf-tcpm-rfc8312bis.md [hunk 1]
```diff
@@ -982,14 +986,14 @@ independently of the level of statistical multiplexing on the link.
...
-## Behavior for Application-Limited Flows
+## Behavior for Application-Limited Flows {#app-limited}
 
 CUBIC does not increase its congestion window size if a flow is
 currently limited by the application instead of the congestion window.
-In case of long periods during which *cwnd* has not been updated due
-to such an application limit, such as idle periods, *t* in {{eq1}}
-MUST NOT include these periods; otherwise, W<sub>cubic</sub>(*t*)
-might be very high after restarting from these periods.
+{{#win-inc}} requires that *t* in {{eq1}} does not include long
```
Comment (larseggert) [inline draft-ietf-tcpm-rfc8312bis.md:993]:
```suggestion
{{win-inc}} requires that *t* in {{eq1}} does not include long
```

**Comment by goelvidhi** [comment]
Thank you @bbriscoe for working on this. The text change looks good to me.
For every issue, we add Issue history in Appendix B, could you add that?

File: draft-ietf-tcpm-rfc8312bis.md [hunk 2]
```diff
@@ -982,14 +986,14 @@ independently of the level of statistical multiplexing on the link.
...
 CUBIC does not increase its congestion window size if a flow is
 currently limited by the application instead of the congestion window.
-In case of long periods during which *cwnd* has not been updated due
-to such an application limit, such as idle periods, *t* in {{eq1}}
-MUST NOT include these periods; otherwise, W<sub>cubic</sub>(*t*)
-might be very high after restarting from these periods.
+{{win-inc}} requires that *t* in {{eq1}} does not include long
+application-limited periods, such as idle periods, otherwise
+W<sub>cubic</sub>(*t*) might be very high after restarting from these
+periods.
```
Comment (goelvidhi) [inline draft-ietf-tcpm-rfc8312bis.md:997]: Thank you @bbriscoe for working on this. The text change looks good to me.
For every issue, we add Issue history in Appendix B, could you add that?

**Review by markkukojo — CHANGES_REQUESTED** [review]
This is better located now, thanks.

However, the requirement is quite vaguely stated. What is "a long period", i.e., what is the rule what an implementer can use to decide whether or not to include an idle/application-limited period in t? Moreover, an idle period is quite trivial for a TCP sender to detect, but what is the rule for a TCP sender to detect/decide that it does not fully use cwnd? An arbitrary implementer must be able to implement this correctly.

**Comment by larseggert** [comment]
This isn't CUBIC-specific though, is it? What do other CC algorithms say about app-limited periods?

**Review by lisongxu — APPROVED** [review]
Thanks

**Comment by lisongxu** [comment]
It is hard to accurately define how long is long, as intuitively, it depends on the change of cwnd. The goal here is to reduce or limit the sudden big changes of cwnd. It is acceptable to have some minor changes of cwnd after an app-limited period, but not acceptable to have a big change of cwnd. Similarly, it is hard to define how big is a big change of cwnd. 

Maybe we can remove "long"?

Thanks

**Comment by nealcardwell** [comment]
I agree with @lisongxu  that just dropping "long" should be fine, so that we have something like:

+ {{win-inc}} requires that *t* in {{eq1}} does not include
+ application-limited periods, such as idle periods, otherwise
+ W<sub>cubic</sub>(*t*) might be very high after restarting from these
+ periods.

**Pull request merged**